draft-ietf-httpbis-header-structure.md

```diff
@@ -777,7 +781,8 @@ NOTE: This algorithm parses both Integers {{integer}} and Floats {{float}}, and
    6. If type is "float" and input_number contains more than 16 characters, fail parsing.
 8. If type is "integer":
    1. Parse input_number as an integer and let output_number be the product of the result and sign.
-   2. If output_number is outside the range defined in {{integer}}, fail parsing.
+   2. If output_number is not supported accurately by the implementation (as noted in {{integer}}), fail parsing.
+   3. If output_number is outside the range for integers defined in {{integer}}, fail parsing.
```

Comment:
As noted, this needs another condition ...and the implementation does not support integers outside this range.

Reply:
Where?

Reply:
Let's say that I support -2^63 to 2^63-1, something that isn't unreasonable if you aren't javascript.  This code turns into:

```js
if (n < -2^63 || n > 2^63-1) { // Step 2.
  return fail_parsing;
}
if (n < -2^53+1 || n > 2^53-1) { // Step 3.
  return fail_parsing;
}
```

Pass n = 2^60 into that code and you get an error.  Hence my insistence that the first test is adequate.

Now, you could read "the range for integers" as meaning  -2^63 to 2^63-1 in this case, but then you just repeated yourself.

Does that help clarify the concern?